src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -290,6 +294,54 @@ protected function getFormActions()
                     ->setUseButtonTag(true)
                     ->addExtraClass('btn-outline-danger btn-hide-outline font-icon-trash-bin action--delete'));
             }
+
+            if ($gridStateStr = ($this->getRequest()->getVar('gridState')
+                || $this->getRequest()->postVar('gridState'))) {
+                $this->gridField->getState(false)->setValue($gridStateStr);
+                $this->owner->gridField->getState(false)->setValue($gridStateStr);
+            }
+
+            $actions->push(HiddenField::create('gridState', null, $this->getRequest()->getVar('gridState')));
+
```

Comment:
The initial check is against POST or GET, but this is only GET. Shouldn't this just be using `$gridStateStr`?